devicetypes/stelpro/stelpro-maestro-thermostat.src/stelpro-maestro-thermostat.groovy

```diff
@@ -326,6 +302,45 @@ def parse(String description) {
 	return result
 }
 
+def handleTemperature(descMap) {
+	def map = [:]
+	def intVal = Integer.parseInt(descMap.value, 16)
+	map.name = "temperature"
+	map.unit = getTemperatureScale()
+	map.value = getTemperature(descMap.value)
+
+	// Handle special temperature flags where we need to change the event type
+	if (intVal == 0x7ffd) {
```

Comment:
Here, in order to take care of the bug in the device, I would simply rearrange the if-else statement to check for a negative value at the end.

If a negative temperature value is found (not 0x8000, which is an error code), then the Freeze alarm should be set.